subprojects/launcher/src/main/java/org/gradle/launcher/exec/InProcessBuildActionExecuter.java

```diff
@@ -48,9 +52,17 @@ public Object execute(BuildAction action, BuildRequestContext buildRequestContex
                 return buildController.getResult();
             } finally {
                 buildLifecycleListener.beforeComplete();
+                renderDeprecationReport(gradleLauncher);
             }
         } finally {
             buildController.stop();
         }
     }
+
+    private void renderDeprecationReport(GradleLauncher launcher) {
+        ProjectInternal project = launcher.getGradle().getDefaultProject();
```

Comment:
❌ We should log in the root project, not the default project.

Reply:
Done. But sometimes rootProject is not available (e.g. init script failure) and `IllegalStatementException` would be thrown. In this case, we fall back to use default project.

Reply:
I don't think we need a deprecation report if we can't even compute the root project.

Reply:
Sure, done.